Fix easy mypy errors

`mypy yamm` shows a few errors, but two sets of errors are easily resolvable

Errors in `class Trace`
```
yamm/constructs/trace.py:45: error: Name "coords" already defined on line 20
yamm/constructs/trace.py:68: error: Name "crs" already defined on line 21
```
since `coords` and `crs` are defined as properties, the declaration at the top of the class are an unneeded extra definitions

Errors in `plot_map()`
```
yamm/utils/plot.py:124: error: "MapInterface" has no attribute "g"
yamm/utils/plot.py:126: error: "MapInterface" has no attribute "_geom_key"
yamm/utils/plot.py:126: error: "MapInterface" has no attribute "crs"
```
`plot_map()` is a utility function which is only used in the examples. its type hint is `MapInterface` but the implementation is specific to `NxMap`. Given the function's limited usage, this type hint can be changed to the specific type.

--- a/yamm/constructs/trace.py
+++ b/yamm/constructs/trace.py
@@ -17,9 +17,6 @@ from yamm.utils.geohash import encode
 class Trace:
     _frame: GeoDataFrame
 
-    coords: List[Coordinate]
-    crs: CRS
-
     def __init__(self, frame: GeoDataFrame):
         self._frame = frame
 

--- a/yamm/utils/plot.py
+++ b/yamm/utils/plot.py
@@ -8,7 +8,7 @@ from shapely.geometry import Point
 
 from yamm.constructs.match import Match
 from yamm.constructs.trace import Trace
-from yamm.maps.map_interface import MapInterface
+from yamm.maps.nx.nx_map import NxMap
 from yamm.utils.crs import XY_CRS, LATLON_CRS
 
 
@@ -43,7 +43,7 @@ def plot_trace(trace, m=None, point_color="yellow", line_color="green"):
     return m
 
 
-def plot_matches(matches: List[Match], road_map: MapInterface):
+def plot_matches(matches: List[Match], road_map: NxMap):
     """
     plots a trace and the relevant matches on a folium map
 
@@ -81,7 +81,7 @@ def plot_matches(matches: List[Match], road_map: MapInterface):
         }
 
         return d
-    
+
 
     road_df = pd.DataFrame([match_to_road(m) for m in matches if m.road])
     road_df = road_df.loc[road_df.road_id.shift() != road_df.road_id]
@@ -119,7 +119,7 @@ def plot_matches(matches: List[Match], road_map: MapInterface):
     return fmap
 
 
-def plot_map(tmap: MapInterface, m=None):
+def plot_map(tmap: NxMap, m=None):
     # TODO make this generic to all map types, not just NxMap
     roads = list(tmap.g.edges(data=True))
     road_df = pd.DataFrame([r[2] for r in roads])

--- a/yamm/utils/plot.py
+++ b/yamm/utils/plot.py
@@ -2,12 +2,10 @@ from typing import List, Optional
 
 import folium
 import geopandas as gpd
-import numpy as np
 import pandas as pd
 from shapely.geometry import Point
 
 from yamm.constructs.match import Match
-from yamm.constructs.trace import Trace
 from yamm.maps.nx.nx_map import NxMap
 from yamm.utils.crs import XY_CRS, LATLON_CRS
 

